pandas/core/arrays/base.py

```diff
@@ -511,7 +513,15 @@ def nbytes(self) -> int:
     # Additional Methods
     # ------------------------------------------------------------------------
 
-    def astype(self, dtype, copy=True):
+    @overload
+    def astype(self, dtype: NpDtype, copy: bool = True) -> np.ndarray:
```

Comment:
dt64/td64 generally go to DTA/TDA

Reply:
Don't think so.  When you pass a numpy dtype, you get a `np.ndarray`:
```
>>> a=pd.array([np.datetime64("2021-03-15"), np.datetime64("2021-04-05")])
>>> a
<DatetimeArray>
['2021-03-15 00:00:00', '2021-04-05 00:00:00']
Length: 2, dtype: datetime64[ns]
>>> b=a.astype(np.datetime64)
>>> b
array(['2021-03-15T00:00:00.000000000', '2021-04-05T00:00:00.000000000'],
      dtype='datetime64[ns]')
>>> type(b)
<class 'numpy.ndarray'>
```